### ISSUE TITLE
How to know if Insertion was successful or not

### ISSUE THREAD

**Issue by vivekkartha** [satyan/sugar#222]
How can we know if an insert operation was successful into a column with @Unique constrain?

**Comment by jivimberg** [comment]
I may be wrong, but I think that in case of conflict the used policy is to replace the row so the insert doesn't fail in such case.

**Comment by vivekkartha** [comment]
There are cases where I want the insertion to fail and notify the user about it on encountering a duplicate.

**Comment by satyan** [comment]
You may want to modify the save method to take in the conflict resolution policy.
db.insertWithOnConflict(NamingHelper.toSQLName(object.getClass()), null, values,
                SQLiteDatabase.CONFLICT_REPLACE);

You could use Fail on error policy if that suits your needs.

We could think about taking the policy as an input.

**Comment by vivekkartha** [comment]
Yes, taking the policy as an input would help a great deal.  And maybe return a boolean based on the success/failure of the insertion.
